### PR TITLE
Testing infrastructure skip if simulators not available

### PR DESCRIPTION
**Pull request by mkatsimpris**
Added --sim option for the tests. Also the default is "ghdl". travis.yml changed to run the test with ghdl and iverilog. if the ghdl simulator is not installed then the conversion test is passed.
@cfelton

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -43,7 +43,8 @@ install:
 
 script:
    - cd test
-   - coverage run --source=jpegenc -m py.test --include-reference
+   - coverage run --source=jpegenc -m py.test --sim=ghdl 
+   - coverage run --source=jpegenc -m py.test --sim=iverilog
```
Comment (cfelton) [inline .travis.yml]: I still want the travis-ci to run the reference design cosimulation `--include-reference` and that will only run with the Verilog simulators.

File: test/conftest.py [hunk 2]
```diff
@@ -1,15 +1,40 @@
...
+
+xfail = pytest.mark.xfail
+
+all_sims = list(_simulators)
+
+if sys.version_info[0] > 2:
+    collect_ignore = ['conversion/toVerilog/test_not_supported_py2.py']
 
 def pytest_addoption(parser):
-    parser.addoption("--include-reference", action="store_true",
```
Comment (cfelton) [inline test/conftest.py]: Why was this option removed?

File: test/test_dct_1d.py [hunk 3]
```diff
@@ -14,7 +14,9 @@
 from jpegenc.testing import sim_available, run_testbench
 from jpegenc.testing import clock_driver, reset_on_start, pulse_reset
 
-simsok = sim_available('iverilog') and sim_available('ghdl')
+simsok = sim_available('ghdl')
```
Comment (cfelton) [inline test/test_dct_1d.py]: Why only use `ghdl`?

File: test/test_dct_1d.py [hunk 4]
```diff
@@ -14,7 +14,9 @@
...
+"""default simulator"""
+verify.simulator = "ghdl"
```
Comment (cfelton) [inline test/test_dct_1d.py]: Should this use the "sim" defined at the command line?

File: test/conftest.py [hunk 5]
```diff
@@ -1,15 +1,40 @@
...
-                     help="Include the reference design cosim tests")
+    parser.addoption("--sim", action="store", choices=all_sims,
+                     help="HDL Simulator")
 
 
 def pytest_configure(config):
-    sys._called_from_test = True
```
Comment (cfelton) [inline test/conftest.py]: Why was this removed?

File: test/test_dct_1d.py (hunk 4 above)
Comment (cfelton) [inline test/test_dct_1d.py]: Can the default simulator be sent somewhere global, instead of each file?

File: test/conftest.py [hunk 6]
```diff
@@ -1,15 +1,40 @@
...
+    sim = config.getoption('sim')
+    if sim is not None:
+        verify.simulator = analyze.simulator = sim
```
Comment (cfelton) [inline test/conftest.py]: The default simulator should be set here, if present, then it doesn't need to be set in each file.  Also, the check if the simulator is present.

A new decorator could be made so the "simok" check doesn't need to be made in each test, `@skipif_no_sim`

File: test/test_dct_1d.py (hunk 4 above)
Comment (mkatsimpris) [inline test/test_dct_1d.py]: Ok. It is better to set a default simulator in a file.

File: .travis.yml (hunk 1 above)
Comment (mkatsimpris) [inline .travis.yml]: Ok. I will fix it.

File: test/conftest.py (hunk 5 above)
Comment (mkatsimpris) [inline test/conftest.py]: It is a mistake. I will replace the lines
